sections/section_util.go

```diff
@@ -152,231 +95,16 @@ func NewCommonUSGPCSegment(bs *util.BitStream) (CommonUSGPCSegment, error) {
-	}
-
-	uspnatCore.SaleOptOutNotice, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SaleOptOutNotice", err)
-	}
-
-	uspnatCore.SharingOptOutNotice, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SharingOptOutNotice", err)
-	}
-
-	uspnatCore.TargetedAdvertisingOptOutNotice, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.TargetedAdvertisingOptOutNotice", err)
-	}
-
-	uspnatCore.SensitiveDataProcessingOptOutNotice, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SensitiveDataProcessingOptOutNotice", err)
-	}
-
-	uspnatCore.SensitiveDataLimitUseNotice, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SensitiveDataLimitUseNotice", err)
-	}
-
-	uspnatCore.SaleOptOut, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SaleOptOut", err)
-	}
-
-	uspnatCore.SharingOptOut, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SharingOptOut", err)
-	}
-
-	uspnatCore.TargetedAdvertisingOptOut, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.TargetedAdvertisingOptOut", err)
-	}
-
-	uspnatCore.SensitiveDataProcessing, err = bs.ReadTwoBitField(12)
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.SensitiveDataProcessing", err)
-	}
-
-	uspnatCore.KnownChildSensitiveDataConsents, err = bs.ReadTwoBitField(2)
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.KnownChildSensitiveDataConsents", err)
-	}
-
-	uspnatCore.PersonalDataConsents, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.PersonalDataConsents", err)
-	}
-
-	uspnatCore.MspaCoveredTransaction, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.MspaCoveredTransaction", err)
-	}
-
-	uspnatCore.MspaOptOutOptionMode, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.MspaOptOutOptionMode", err)
-	}
-
-	uspnatCore.MspaServiceProviderMode, err = bs.ReadByte2()
-	if err != nil {
-		return uspnatCore, errorHelper("CoreSegment.MspaServiceProviderMode", err)
-	}
-
-	return uspnatCore, nil
-}
-
-func NewUPSUTCoreSegment(bs *util.BitStream) (USPUTCoreSegment, error) {
-	var usputCore USPUTCoreSegment
-	var err error
-
-	usputCore.Version, err = bs.ReadByte6()
-	if err != nil {
-		return usputCore, errorHelper("CoreSegment.Version", err)
-	}
-
-	usputCore.SharingNotice, err = bs.ReadByte2()
-	if err != nil {
-		return usputCore, errorHelper("CoreSegment.SharingNotice", err)
-	}
-
-	usputCore.SaleOptOutNotice, err = bs.ReadByte2()
+	commonUSGPC.SubsectionType, err = bs.ReadByte2()
 	if err != nil {
-		return usputCore, errorHelper("CoreSegment.SaleOptOutNotice", err)
+		return commonUSGPC, ErrorHelper("GPCSegment.SubsectionType", err)
 	}
 
```

Comment:
Should we also check that SubsectionType is `1`? According to the spec it seems to be the only allowed value.